### ISSUE TITLE
mimirtool not throwing logs when hitting timeouts

### ISSUE THREAD

**Issue by achetronic** [grafana/mimir#5229]
#### Describe the bug
when executing mimirtool with an unreachable address (for example, setting https when only http is available) does not produce logs even when debug mode is set

This is a bad behaviour because it hides whan happens behind. For example using mimirtool Docker image, being executed but not causing success nor error until a long time. After a while it exits with error but no reason thrown

#### Expected behavior

When an address is simply unreachable and mimirtool hits timeout, simply throw a log with that before the retry

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: Bare process execution

**Comment by dimitarvdimitrov** [comment]
This is the code of the client that mimirtool uses to reach mimir. I see an error line with "error during request to Grafana Mimir API"

https://github.com/grafana/mimir/blob/ea4bca1e8603543902e7da8b5a20994ed6117d3e/pkg/mimirtool/client/client.go#L155-L159

Can you reproduce this with the binary as well as the Docker image @achetronic ?

**Comment by achetronic** [comment]
hello @dimitarvdimitrov 

Yes, if you set an https address and this address can not be resolved, reached, etc, the behaviour is the same. I tested it with the binary to discard possible Docker image building issues from the beginning :)

**Comment by achetronic** [comment]
@dimitarvdimitrov No, no, just hang until timeout. This is why docker image simply get stuck with no output for a loooong and then exit errored. Basically, it hangs until some kind of timeout

**Comment by dimitarvdimitrov** [comment]
is this timeout something that GitHub actions enforces or a timeout the mimirtool enforces? Can you share some output from such a failed run?

**Comment by achetronic** [comment]
> is this timeout something that GitHub actions enforces or a timeout the mimirtool enforces? Can you share some output from such a failed run?

Hey, no Github Actions here, just Docker container inside K8s not throwing any kind of logs on the sittuation I mentioned. After a while, container exit errored.

After that, I tested exactly the same with the binary directly on the machine and the same happens, so it is a problem of the binary.

If I find a slot, I will try to debug the code and open a PR, as for sure it is related to it :)
